lib/cloud-metadata/gcp.js

```diff
@@ -78,76 +96,37 @@ function getMetadataGcp(
 /**
  * Builds metadata object
  *
- * Takes the response from a /computeMetadata/v1/?recursive=true
- * service request and formats it into the cloud metadata object
+ * Convert a GCP Cloud Engine VM metadata response
+ * (https://cloud.google.com/compute/docs/metadata/default-metadata-values)
+ * to the APM intake cloud metadata object
+ * (https://github.com/elastic/apm/blob/main/specs/agents/metadata.md#gcp-metadata).
+ *
+ * See discussion about big int values here:
+ * https://github.com/googleapis/gcp-metadata#take-care-with-large-number-valued-properties
+ * This implementation is using the same 'json-bigint' library as 'gcp-metadata'.
  */
 function formatMetadataStringIntoObject(string) {
-  const data = JSON.parse(string);
-  // cast string manipulation fields as strings "just in case"
-  if (data.instance) {
-    data.instance.machineType = String(data.instance.machineType);
-    data.instance.zone = String(data.instance.zone);
-  }
+  const data = JSONBigInt.parse(string);
+
+  // E.g., 'projects/513326162531/zones/us-west1-b' -> 'us-west1-b'
+  const az = data.instance.zone.split('/').pop();
 
   const metadata = {
-    availability_zone: null,
-    region: null,
+    provider: 'gcp',
     instance: {
-      id: null,
+      id: data.instance.id.toString(), // We expect this to be a BigInt.
+      name: data.instance.name,
```

Comment:
Reviewer note: This function has been simplified (IMO). The `instance.name` had been missed before (or perhaps the APM spec changed after original impl?).  The `data.instance.id` is the interesting BigInt case.